test/parser/LiteralOrIriTest.cpp

```diff
@@ -278,3 +278,45 @@ TEST(LiteralOrIri, Hashing) {
   ad_utility::HashSet<LiteralOrIri> set{lit, iri};
   EXPECT_THAT(set, ::testing::UnorderedElementsAre(lit, iri));
 }
+
+// _______________________________________________________________________
+TEST(LiteralTest, SetSubstr) {
+  LiteralOrIri literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  literal.getLiteral().setSubstr(0, 5, 12);
+  EXPECT_THAT("Hello", asStringViewUnsafe(literal.getContent()));
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype()));
+
+  literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  literal.getLiteral().setSubstr(6, 5, 12);
+  EXPECT_THAT("World", asStringViewUnsafe(literal.getContent()));
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype()));
+
+  literal = LiteralOrIri::literalWithoutQuotes("Äpfel");
+  literal.getLiteral().setSubstr(0, 2, 6);
+  EXPECT_THAT("Ä", asStringViewUnsafe(literal.getContent()));
+
+  // Test with invalid values.
+  literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  EXPECT_THROW(literal.getLiteral().setSubstr(12, 1, 12),
+               ad_utility::Exception);
+  EXPECT_THROW(literal.getLiteral().setSubstr(6, 7, 12), ad_utility::Exception);
+}
+
+// _______________________________________________________________________
+TEST(LiteralTest, removeDatatypeOrLanguageTag) {
+  LiteralOrIri literal = LiteralOrIri::literalWithoutQuotes(
+      "Hello World!",
+      Iri::fromIriref("<http://www.w3.org/2001/XMLSchema#string>"));
+  literal.getLiteral().removeDatatypeOrLanguageTag();
+  EXPECT_THAT("Hello World!", asStringViewUnsafe(literal.getContent()));
+  EXPECT_FALSE(literal.hasDatatype());
+  EXPECT_THROW(literal.getDatatype(), ad_utility::Exception);
```

Comment:
1. Also test a language tag that is removed.
2. Also test a plain literal without any suffix (where this function should do nothing).